<launch>
    <arg name="comparison_mode"      default="false"/>
    <arg name="base_footprint_name"  default="potential"/>
    <arg name="teleop_topic_name"    default="/potential/input/teleop"/>
    <arg name="odom_topic_name"      default="/potential/odom"/>

    <arg name="weight_obstacle"     default="1.0"/>
    <arg name="weight_goal"         default="3.0"/>
    <arg name="delta"               default="0.05"/>
    <arg name="grid_width"          default="0.12"/>

    <arg name="sensor_size"         default="50"/>

    <arg name="rviz"                    default="true"/>
    <arg name="rviz_cfg"                default="$(find local_path_planning_tutorial)/config/rviz/potential.rviz"/>
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_cfg)"/>

    <!-- leonard_jones_potential -->
    <group ns = "potential_method_path_generator">
        <node pkg="local_path_planning_tutorial" type="potential_method_path_generator" name="potential_method_path_generator" launch-prefix="xterm -font r16 -bg darkblue -e">
            <param name="base_footprint_name"  type="str"     value="$(arg base_footprint_name)"/>
            <param name="teleop_topic_name"    type="str"     value="$(arg teleop_topic_name)"/>
            <param name="odom_topic_name"      type="str"     value="$(arg odom_topic_name)"/>
            <param name="weight_obstacle"      type="double"  value="$(arg weight_obstacle)"/>
            <param name="weight_goal"          type="double"  value="$(arg weight_goal)"/>
            <param name="delta"                type="double"  value="$(arg delta)"/>
            <param name="grid_width"           type="double"  value="$(arg grid_width)"/>
        </node>
    </group>

    <group ns = "virtual_environment">
        <node pkg="local_path_planning_tutorial" type="ppac_robot_pose_broadcaster" name="potential_robot_pose_broadcaster">
            <param name="base_footprint_name"  type="str"     value="$(arg base_footprint_name)"/>
            <param name="teleop_topic_name"    type="str"     value="$(arg teleop_topic_name)"/>
            <param name="odom_topic_name"      type="str"     value="$(arg odom_topic_name)"/>
        </node>
        
        <node unless="$(arg comparison_mode)" pkg="local_path_planning_tutorial" type="ppac_virtual_environment" name="virtual_environment">
            <param name="sensor_size"          type="int"     value="$(arg sensor_size)"/>
        </node>
    </group>
</launch>

